pkg/agent/entrypoint.go

```diff
@@ -79,18 +75,31 @@ func (r MainRunner) Run(logger *zap.Logger, ctx context.Context) error {
 		}
 	}
 
-	logger.Info("Entering main loop")
-	for {
-		event, err := vmEventQueue.Wait(ctx)
-		if err != nil {
-			if ctx.Err() != nil {
-				// treat context canceled as a "normal" exit (because it is)
-				return nil
-			}
+	mc, err := billing.NewMetricsCollector(ctx, logger, &r.Config.Billing)
+	if err != nil {
+		return fmt.Errorf("error creating billing metrics collector: %w", err)
+	}
 
-			logger.Error("vmEventQueue returned error", zap.Error(err))
-			return err
+	tg := taskgroup.NewGroup(logger, taskgroup.WithParentContext(ctx))
+	tg.Go("collect-metrics", func(logger *zap.Logger) error {
+		return mc.Run(tg.Ctx(), logger, storeForNode, metrics)
+	})
```

Comment:
A couple questions here:

1. Will this change the value of `.logger` in the billing-related log lines? (and if so, from what to what?)
2. Are the billing client goroutines now using taskgroup as well?

Reply:
1. It will expand "logger" field in the json logline by more more segment - "collect-metrics"
2. You mean the onces which are run from the `MetricsCollector`? No, they are still `go ...()`.

Reply:
So AFAICT then, the billing client log lines will have `logger` fields like `autoscaler-agent.collect-metrics.send-http` -- and collection will have `autoscaler-agent.collect-metrics.collect` ?

IMO:

1. They should still mention `billing` explicitly
2. Client `logger`s should not mention "collect-metrics" because that's a separate process
3. We shouldn't have duplicated `collect-metrics.collect`

Reply:
Changed to `billing-metrics` in 954e2c6. Does this really matter in practice though, e.g. do you even look at the logger name?

Reply:
I often use it for filtering -- e.g. to show all errors from `autoscaler-agent.billing.send-http`, or something. It's nice to keep those consistent.